### PR TITLE
Developer guide - enroll feature is added

### PR DESCRIPTION
**Pull request by michael-lee-sk**

### CODE REVIEW

**Comment by muhammad-khair** [comment]
For your object diagrams, maybe change the second one to `Final State`?

image

Other than that seems good!

File: docs/diagrams/EnrollLessonState1.puml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@startuml
+
+title Initial state
```
Comment (eezj35) [inline docs/diagrams/EnrollLessonState1.puml:3]: I think this should be final?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -184,19 +184,79 @@ Command syntax
...
 
-Object diagram
+The Enroll feature establishes a bi-directional relationship
+between a student and a specific lesson. This student to be enrolled must meet a set of conditions.
+1. The student has to have the same `Grade` as specified in the lesson.
+2. The student cannot already be enrolled in the lesson.
 
-Sequence diagram
+When enrolled, the student will have that lesson added to their list
+of existing lessons, while the lesson size would increase by 1.
```
Comment (eezj35) [inline docs/DeveloperGuide.md:195]: maybe can say list of lessons signed up for/ lessons enrolled, something along those lines

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -184,19 +184,79 @@ Command syntax
...
-Activity diagram
+Given below is an example usage scenario and how the enroll operation behaves.
+_Note: For this usage, we only consider the main success scenario
+(i.e. the student exists, has the same `grade` as the specific lesson,
+and has not been enrolled in the lesson yet)._
+
+<ins>Step 1:</ins>
+
+User has a list of students and lessons presented in their
+TuitiONE application. The user has one has a `Lesson` with the lessoncode Math-P2-Wed-1800 and `grade`
```
Comment (eezj35) [inline docs/DeveloperGuide.md:205]: i think there might be some grammatical errors in this sentence

**Review by eezj35 — APPROVED** [review]
Overall good job! Only a few minor tweaks. Thanks for the edits!

**Pull request merged**